rpc_server.go

```diff
@@ -334,6 +334,61 @@ func (s *SourceControl) WriteComment(comment *string, reply *bool) error {
 	return nil
 }
 
+// CouplingStatus describes the status of FB / error coupling
+type CouplingStatus int
+
+const (
+	NoCoupling CouplingStatus = iota + 1 // FB and error aren't coupled
+	FBToErr                              // FB triggers cause secondary triggers in error channels
+	ErrToFB                              // Error triggers cause secondary triggers in FB channels
+)
+
+// CoupleErrToFB turns on or off coupling of Error -> FB
+func (s *SourceControl) CoupleErrToFB(couple *bool, reply *bool) error {
+	s.mu.Lock()
```

Comment:
should you check for source == nil before getting lock?

Reply:
Wont this have the same race condition issues as SetMix, and thus need the same solution (eg a go routine?), also the function that grabs the lock should be the same, which I think means it should be in lancero.Lancero.SetCoupling.

Reply:
At first I agreed, but no. It would be a race condition to check for `s.activeSource==nil` without the lock, because a different RPC client might request Start or Stop commands concurrently. These will change the `s.activeSource ` value. And the lock that's being held here is s's lock.

Another difference is that SetMix needs to hold the _source's_ lock to proceed, while changing group trigger needs to hold only the trigger broker's lock. The source's lock is hard to get, because `blockingRead` holds it for most of the time and then tries to grab it again as soon as it lets it go. The broker's lock, by contrast, is normally held for a small fraction of the time (only while primary triggers are being checked for secondary triggers).

I don't think LanceroSource.SetCoupling needs a lock, because it's not manipulating any `LanceroSource` variables. It's just calling a method of the trigger broker, which itself uses a lock.

Anyway, I think the issue here is confusion between the locks that lock `SourceControl` and those that lock an `AnySource`. I'm going to stand by this way of doing it.

Reply:
Thanks for the explanation, I buy it.